warplib/src/warp_memory/arrays.cairo

```diff
@@ -83,8 +82,7 @@ impl WarpMemoryArraysImpl of WarpMemoryArraysTrait {
              panic_with_felt252('Index out of range');
         } 
 
-        let index_location = array_ptr + index * elem_width;
-        self.read(index_location) 
+        array_ptr + index * elem_width
```

Comment:
Why the change to return absolute pointer instead of the value itself?

Reply:
Yeah, that was actually an error on my side. The index function returns the address of the "object" you want to read. What it was doing was getting the address and then reading from it which is a wrong behaviour since it should only get it and let other functions (such as accessors) read from that position